src/components/encoding-pane/property-editor.tsx

```diff
@@ -45,16 +44,37 @@ export class PropertyEditorBase extends React.PureComponent<PropertyEditorProps,
 
   protected changeFieldProperty(result: any) {
     const {prop, nestedProp, shelfId, handleAction} = this.props;
-    const value = result.formData;
-    handleAction({
-      type: SPEC_FIELD_NESTED_PROP_CHANGE,
-      payload: {
-        shelfId,
-        prop,
-        nestedProp,
-        value
-      }
-    });
+    const value = result.formData[Object.keys(result.formData)[0]];
+    if (nestedProp) {
+      handleAction({
+        type: SPEC_FIELD_NESTED_PROP_CHANGE,
+        payload: {
+          shelfId,
+          prop,
+          nestedProp,
+          value
+        }
+      });
+    } else {
+      handleAction({
+        type: SPEC_FIELD_PROP_CHANGE,
+        payload: {
+          shelfId,
+          prop,
+          value
+        }
+      });
+    }
+  }
+
+  private getFormData() {
+    const {fieldDef} = this.props;
+    return {
+      'ScaleType': fieldDef.scale ? fieldDef.scale.type : undefined,
```

Comment:
1) This is still manually typed, which is tedious and error prone.
2) I believe this will depend on field type, just like `getFieldPropertyGroupIndex()`.

Since you have all the information needed from [`getFieldPropertyGroupIndex()`]( https://github.com/vega/voyager/pull/789/files#diff-6e0fc12971841df1d666cfddd2fb434dR139),  I think you can extract `getFieldPropertyGroupIndex()` to be a part of the schema file and use it both here and in `field-customizer.tsx`.